src/internal/connector/discovery/api/users.go

```diff
@@ -184,17 +195,112 @@ func (c Users) GetInfo(ctx context.Context, userID string) (*UserInfo, error) {
 		userInfo = newUserInfo()
 	)
 
-	// TODO: OneDrive
-	_, err = c.stable.Client().UsersById(userID).MailFolders().Get(ctx, nil)
+	err = c.allowsExchange(ctx, userID, userInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.allowsOnedrive(ctx, userID, userInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.getAdditionalData(ctx, userID, userInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return userInfo, nil
+}
+
+// verify mailbox enabled for user
+func (c Users) allowsExchange(ctx context.Context, userID string, userInfo *UserInfo) error {
```

Comment:
This function leans a bit too heavily towards the interpretation of its results.  What it's fundamentally doing is retrieving exchange mailfolders.  We want to eventually interpret the results as proof the user can access exchange.  But if we needed to make a different interpretation based on mailfolders, we'd have to change this func signature or create a new func, right?  Packing the interpretation into the func itself makes it less portable.

Instead of adding a set of `allowsFoo()` funcs, it'd be better to add thin wrappers on each call: getUsersMailFolders, getUsersDrives, getMailboxSettings, etc.  The caller can interpret the results however they want.

Reply:
done

Reply:
Ah, I don't think I explained my expectations clearly enough, since these functions are still of the `allowsFoo` pattern.

API functions should avoid interpreting a result when they could return that result instead.  That means, we don't want `allowsFoo`.  Rather, we want `getFoo`.  The caller can decide whether the result implies whether things are allowed or not (or, alternatively, we can have a second function which centralizes that evaluation).

So this shouldn't be `allowsExchange`.  It should be `GetMailboxes`, and it should return the mailboxes it fetched, and an error.